components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -2062,4 +2065,76 @@ private void sendEmailNotification(String apiId, String apiName, String newVersi
             }
         }
     }
+
+    /**
+     * Discover and return a list of service endpoints
+     *
+     * @return {@code List<Endpoint>}
+     * @throws APIManagementException If an error occurred while discovering services
+     */
+    @Override
+    public List<Endpoint> discoverServiceEndpoints() throws APIManagementException {
+        List<Endpoint> discoveredEndpointList = new ArrayList();
+        try {
+            ServiceDiscoveryConfigurations serviceDiscoveryConfig = ServiceDiscoveryConfigBuilder
+                    .getServiceDiscoveryConfiguration();
+
+            // If service discovery not enabled - do not proceed
+            if (!serviceDiscoveryConfig.isServiceDiscoveryEnabled()) {
+                log.error("Service Discovery not enabled");
+                return discoveredEndpointList;
+            }
+            List<ServiceDiscoveryImplConfig> implConfigList = serviceDiscoveryConfig.getImplementationsList();
+            for (ServiceDiscoveryImplConfig implConfig : implConfigList) {
+                //Every implConfig has two elements. The implClass and the implParameters.
+
+                /* Get the implClass instance */
+                String implClassName = implConfig.getImplClass();
+                Class implClazz = Class.forName(implClassName);
```

Comment:
Should this be "implClass" instead of "implClazz" ?

Reply:
It has to be implClass. Previously it was clazz, which was a generated code. I'll fix it.